src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -283,11 +356,30 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		return nil
 	}
 
-	if *r.pandaCluster.Spec.Replicas > 1 {
-		r.logger.Info("Put broker into maintenance mode",
+	if *r.pandaCluster.Spec.Replicas == 1 {
+		r.logger.Info("Changes in Pod definition other than activeDeadlineSeconds, configurator and Redpanda container name. Deleting pod",
```

Comment:
This if statement doesn't make sense to me. You can remove this altogether. 

If this is special case when replication is equal 1 then 

https://github.com/redpanda-data/redpanda/blob/ba8ddbbc43d3486f6f2b069191eace789f449e01/src/go/k8s/pkg/resources/statefulset_update.go#L366-L367

This needs to return, right?

Reply:
In other words if replication is equal 1 will this logic call maintenance mode? 

If yes, then it's a bug.

Reply:
If replicas == 1 it will skip maintenance mode. This hasn't changed.